addons/trenches/functions/fnc_removeTrench.sqf

```diff
@@ -16,32 +17,41 @@
  * Public: No
  */
 
-params ["_trench", "_unit"];
-TRACE_2("removeTrench",_trench,_unit);
+params ["_trench", "_unit", ["_switchingDigger", false, [true]]];
+TRACE_2("removeTrench", _trench, _unit, _switchingDigger);
 
 private _actualProgress = _trench getVariable [QGVAR(progress), 0];
-if(_actualProgress == 0) exitWith {};
+if (_actualProgress <= 0) exitWith {};
 
 // Mark trench as being worked on
 _trench setVariable [QGVAR(digging), true, true];
+_trench setVariable [QGVAR(diggingType), "DOWN", true];
+private _diggerCount = count (_trench getVariable [QGVAR(diggingPlayers), []]);
 
-private _removeTime = getNumber (configFile >> "CfgVehicles" >> (typeof _trench) >> QGVAR(removalDuration));
-private _removeTimeLeft = _removeTime * _actualProgress;
+if (_diggerCount > 0) then {
+    if !(_switchingDigger) then {
+        [_trench, _unit] call FUNC(addDigger);
+    };
+} else {
+    _trench setVariable [QGVAR(diggingPlayers), [ACE_player], true];
+};
```

Comment:
This is never going to work in MP reliably. Two parties could set this variable at the same time, because they haven
t yet received each others `QGVAR(diggingPlayers)` variable. This will cause both machines to have this array hve one element with the avatar of the opposing unit. If you're going to push to public arrays, you always need to do this on the server machine, to guarantee some kind of order.

Reply:
Reworked the system to use a server as setter